### ISSUE TITLE
Hash is not working properly for interpolated template strings

### ISSUE THREAD

**Issue by mufasa71** [mayank99/ecsstatic#16]
Given:

```css
const radioStyle = css`
  position: absolute;
  top: 0;
  left: 0;
  height: 16px;
  width: 16px;
  background-color: #fff;
  border-radius: 12px;
  border: solid 1px ${Colors.Stone["--stone-2"]};
  cursor: pointer;
`;

const inputStyle = css`
  position: absolute;
  opacity: 0;
  cursor: pointer;
  height: 0;
  width: 0;
  &:checked ~ ${radioStyle}:after {
    content: " ";
    background-color: ${Colors.Blue["--blue-6"]};
    height: 8px;
    width: 8px;
    left: 4px;
    top: 4px;
    position: absolute;
    border-radius: 12px;
  }
`;

const nameStyle = css`
  flex: 2;
  position: relative;

  &:has(${radioStyle}) {
    padding-left: 26px;
  }
`;
```

Output:

```css
.orders-widget-ykv3x2 {
  position: absolute;
  top: 0;
  left: 0;
  height: 16px;
  width: 16px;
  background-color: #fff;
  border-radius: 12px;
  border: solid 1px #a0a0a0;
  cursor: pointer;
}
.orders-widget-1btgk0y {
  position: absolute;
  opacity: 0;
  cursor: pointer;
  height: 0;
  width: 0;
}
.orders-widget-1btgk0y:checked ~ :where(.\1f388-1ix0h6h):after {
  content: " ";
  background-color: #20419a;
  height: 8px;
  width: 8px;
  left: 4px;
  top: 4px;
  position: absolute;
  border-radius: 12px;
}
.orders-widget-1rf7qsr {
  flex: 2;
  position: relative;
}
.orders-widget-1rf7qsr:has(:where(.\1f388-1ix0h6h)) {
  padding-left: 26px;
}
```

1. It does't put proper `classNamePrefix` `\1f388` instead of `orders-widget` in `:where()` selectors
2. Hash generated is wrong in `:where()`selectors,    `.\1f388-1ix0h6h` instead of `.orders-widget-ykv3x2`

**Comment by mayank99** [comment]
thanks for the issue. i think i have an idea why this is happening. i'll look into it later this week

do you mind providing a repro in the meantime? a stackblitz link would be very helpful 🙏

**Comment by mayank99** [comment]
Should be fixed in 0.6.1
